### ISSUE TITLE
Improper handling of JsonNumber

### ISSUE THREAD

**Issue by mm-kk-experiments** [apollographql/apollo-kotlin#4246]
**Summary**
`JsonNumber` is serialized using output of its `toString` method.

**Version**
3.3.2

**Description**
I'm using a custom scalar in my GraphQL API; as I result, I'm passing one of the input values to call the mutation as `com.apollographql.apollo3.api.json.JsonNumber` (I'm using it to avoid loosing the precision while using `Double`). Unfortunately, I noticed that in the output JSON this value is represented in standard format of Java's default `toString` implementation (with class name and hashcode). I started digging into Apollo Kotlin library's code and I found that in `BufferedSinkJsonWriter` we have following line:
```
override fun value(value: JsonNumber) = jsonValue(value.toString())
```
I suppose this is **probably** a simple copy-paste mistake (because `JsonNumber` doesn't have `toString` overridden and it uses default implementation from `Object`/`Any`, which prints only class name and its hashcode). I think the above line should look like this:
```
override fun value(value: JsonNumber) = jsonValue(value.value)
```

Please kindly take a look on it and double-check, thank you.

CC @martinbonnin

**Comment by martinbonnin** [comment]
Yikes, good catch! Want to open a pull request?  And we'll include this in next release (currently planned on Monday).

**Comment by mm-kk-experiments** [comment]
Thanks for workaround, in the meantime I prepared draft PR: https://github.com/apollographql/apollo-kotlin/pull/4247

**Comment by martinbonnin** [comment]
Fixed with https://github.com/apollographql/apollo-kotlin/pull/4247

**Comment by martinbonnin** [comment]
Released in 3.4.0
